scala3doc/src/dotty/dokka/tasty/TypesSupport.scala

```diff
@@ -102,9 +102,10 @@ trait TypesSupport:
       case AnnotatedType(tpe, _) =>
         inner(tpe)
       case tl @ TypeLambda(params, paramBounds, resType) =>
-        // println(params)
-        // println(paramBounds)
-        texts("[") ++ commas(params.zip(paramBounds).map( (name, typ) => texts(s"${name}") ++ inner(typ) )) ++ texts("]")
+        texts("[") ++ commas(params.zip(paramBounds).map { (name, typ) =>
+          val normalizedName = name.takeWhile(_ != '$')
```

Comment:
`$` can be part of appliedType, `class A[+CC <: ([A$B] =>> Any)]` compiles

Reply:
Changed to regex